doc/python/categorical-colors.md

```diff
@@ -0,0 +1,231 @@
+
+### Categorical Color with Plotly Express
+
+Most Plotly Express functions accept a `color` argument which automatically assigns data values to categorical colors **if the data is non-numeric**. If the data is numeric, the color will automatically be considered [continuous](/python/colorscales/). This means that numeric strings must be parsed to be used for continuous color, and conversely, numbers used as category codes must be converted to strings.
+
+For example, in the `tips` dataset, the `smoker` column contains strings:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+fig = px.scatter(df, x="total_bill", y="tip", color="smoker", title="String 'smoker' values mean categorical colors")
+
+fig.show()
+```
+
+The `size` column, however, contains numbers:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+fig = px.scatter(df, x="total_bill", y="tip", color="size", title="Numeric 'size' values mean continous color")
+
+fig.show()
+```
+
+Converting this column to strings is very straightforward, but note that the ordering in the legend is not sequential by default (see below for how to control categorical order):
+
+```python
+import plotly.express as px
+df = px.data.tips()
+df["size"] = df["size"].astype(str)
+fig = px.scatter(df, x="total_bill", y="tip", color="size", title="String 'size' values mean categorical colors")
+
+fig.show()
+```
+
+Converting a string column to a numeric one is also quite straightforward:
+
+```python
+import plotly.express as px
+df = px.data.tips()
+df["size"] = df["size"].astype(str) #convert to string
+df["size"] = df["size"].astype(float) #convert back to numeric
+
+fig = px.scatter(df, x="total_bill", y="tip", color="size", title="Numeric 'size' values mean continous color")
+
+fig.show()
+```
+
+### Color Sequences in Plotly Express
+
+By default, Plotly Express will use the color sequence from the active [template](/python/templates/)'s `layout.colorway` attribute, and the default active template is `plotly` which uses the `plotly` color sequence. You can choose any of the following built-in qualitative color sequences from the `px.colors.qualitative` module, however, or define your own.
+
+```python
+import plotly.express as px
+
+fig = px.colors.qualitative.swatches()
+fig.show()
+```
+
+Color sequences in the `px.colors.qualitative` module are stored as lists of CSS colors:
+
+```python
+import plotly.express as px
+
+print(px.colors.qualitative.Plotly)
+```
+
+Here is an example that creates a scatter plot using Plotly Express, with points colored using the built-in qualitative `G10` color sequence.
+
+```python
+import plotly.express as px
+df = px.data.gapminder()
+fig = px.line(df, y="lifeExp", x="year", color="continent", line_group="country",
+              line_shape="spline", render_mode="svg",
```

Comment:
Can we keep the example short by removing this line? Is it a bug that `spline` gives an error when not using `render_mode='svg'`?

Reply:
I think webgl doesn't support spline, so maybe yes, spline should force svg? maybe too magical... To be honest I just didn't like the look of the figure in webgl :)